src/core/event-handler.js

```diff
@@ -253,7 +276,7 @@ class EventHandler {
      * obj.hasEvent('hello'); // returns false
      */
     hasEvent(name) {
-        return (this._callbacks[name] && this._callbacks[name].length !== 0) || false;
+        return (this._callbacks.has(name) && this._callbacks.get(name).length !== 0) || false;
```

Comment:
```suggestion
        return !!_callbacks.get(nameArg)?.length;
```

Shorter, a method call less, no type error, easier to read?

[playground](https://www.typescriptlang.org/play?strictNullChecks=true&noUnusedLocals=true&exactOptionalPropertyTypes=true&ts=4.6.2&filetype=js#code/PQKhAIAEBcE8AcCm4DeBZAhvAPAZ2gE4CWAdgOYA04GJsA2gLoB8AvuCMAFADGA9ifnAB9bhgA2YgEYZuAa1zgAvOBKIA7uEzwAFAEoA3JxHipM+QDpciaNoBEsRLltU6ARioAmKgGYGBo6IS0nK4ltZ2JLzO4Iz+fALQKhgAtogAggRkSuD2jraGwMDxgiQp6ZnZtpH5nACEtcZBZqFk4aWpGWS6APzmYojk0AAWhgEmwRatNu3lXX0DZMOjQA)

Reply:
`nameArg`?

Reply:
> @willeastcott Sorry, that stemmed from my playground test, because for some reason a variable called "name" triggers errors there.